design/vela-core/componenttrait-composing-and-trait-CR-naming.md

```diff
@@ -0,0 +1,124 @@
+# ComponentTrait Composing and Trait CR Naming
+
+* Owner: Jianbo Sun (@wonderflow), Zhou Zheng Xi (@zzxwill)
+* Reviewers: KubeVela/Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+With the evolution of TraitDefinition, definition name can be short and simple without group and kind information,
+and two TraitDefinitions can both refer to the same Trait CRD. So it's necessary to specify how to compose ComponentTrait
+in ApplicationConfiguration and to set the CR name in a friendly way.
+
+## Several ways to compose ComponentTrait in ApplicationConfiguration
+
+- If the name of TraitDefinition is the same as that of Trait CRD
+
+As shown in [OAM Kubernetes Runtime examples](https://github.com/crossplane/oam-kubernetes-runtime/blob/master/examples/containerized-workload/sample_application_config.yaml), the normal way to compose ComponentTrait is as below.
+
+```
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components:
+    - componentName: example-component
+      traits:
+        - trait:
+            apiVersion: core.oam.dev/v1alpha2
+            kind: ManualScalerTrait
+            spec:
+              replicaCount: 3
+```
+
+- If the name of TraitDefinition is different to that of Trait CRD
+
+The definition name `autoscale` is different to the trait name `autoscalers.standard.oam.dev`, we have two ways to compose ComponentTrait.
+
+```
+apiVersion: core.oam.dev/v1alpha2
+kind: TraitDefinition
+metadata:
+  name: autoscale
+spec:
+  definitionRef:
+    name: autoscalers.standard.oam.dev
+```
+
+1) Use label `trait.oam.dev/type`
+
+```
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components: example-component
+  traits:
+    - trait:
+        apiVersion: standard.oam.dev/v1alpha2
+        kind: Autoscalers
+        metadata:
+          labels:
+            - oam.TraitTypeLabel: "autoscale"
+        spec:
+          replicaCount: 3
+```
+
+When rendering trait, the TraitDefinition could be retrieved by the label.
+
+2) Use `name` field
+
+```
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components: example-component
+  traits:
+    - trait:
+        apiVersion: standard.oam.dev/v1alpha2
+        kind: Autoscalers
+        name: autoscale
```

Comment:
We never have this format. 

refer to https://github.com/oam-dev/spec/blob/master/7.application_configuration.md#trait

`name+properties` or `apiVersion+kind`

No mixing case